src/ui/components/search/searchcomponent.js

```diff
@@ -378,25 +440,27 @@ export default class SearchComponent extends Component {
     if (e && e.relatedTarget) {
       focusStillInSearchbar = this._container.contains(e.relatedTarget);
     }
-    if (this.iconIsFrozen || focusStillInSearchbar) {
+    if (this.iconState === IconState.LOADING || this.iconState === IconState.CUSTOM_LOADING ||
+      this.iconIsFrozen || focusStillInSearchbar) {
       return;
     }
     this.requestIconAnimationFrame(IconState.YEXT);
   }
 
+  oneTimeMouseUpListener () {
+    document.removeEventListener('mouseup', this.onMouseUp);
+    this.iconIsFrozen = false;
+  }
+
   initAnimatedIcon () {
     this.iconState = (this.autoFocus && !this.query) ? IconState.MAGNIFYING_GLASS : IconState.YEXT;
-    this.forwardIcon = DOM.query(this._container, '.js-yxt-AnimatedForward');
-    this.reverseIcon = DOM.query(this._container, '.js-yxt-AnimatedReverse');
     const clickableElementSelectors = ['.js-yext-submit', '.js-yxt-SearchBar-clear'];
     for (const selector of clickableElementSelectors) {
       const clickableEl = DOM.query(this._container, selector);
       if (clickableEl) {
         DOM.on(clickableEl, 'mousedown', () => {
           this.iconIsFrozen = true;
-        });
-        DOM.on(clickableEl, 'mouseup', () => {
-          this.iconIsFrozen = false;
+          DOM.on(document, 'mouseup', this.onMouseUp);
```

Comment:
Yup. If user press down on the icon (which freeze them), and somehow nudge their mouse or lift their finger not on the icon, it would get stuck. It's best to check mouseup on the document instead.